### PR TITLE
Add test for creating and filtering new instance types

### PR DESCRIPTION
**Pull request by adiabramovitch**
Depends on https://github.com/RHEnVision/provisioning-backend/pull/186

### CODE REVIEW

File: internal/clients/new_instance_types_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"context"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/clients/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/aws/smithy-go/ptr"
+	"github.com/stretchr/testify/assert"
+)
```
Comment (lzap) [inline internal/clients/new_instance_types_test.go:13]: Unit tests must not use real clients, only stubs are allowed.

I suggest you to import `test_init.go`:

```
	_ "github.com/RHEnVision/provisioning-backend/internal/testing/initialization"
```

That will do it.

Comment (lzap) [inline internal/clients/new_instance_types_test.go:13]: Was going to recommend importing `_ "github.com/RHEnVision/provisioning-backend/internal/testing/initialization"` but then I realized you only test these functions, nice.

File: internal/clients/new_instance_types_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package clients_test
```
Comment (lzap) [inline internal/clients/new_instance_types_test.go:1]: Can you rename `embed.go` to something reasonable and this file to `something_reasonable_test.go`?

The Idiomatic Go pattern is always keep unit tests named as their source files:

* blah.go
* blah_test.go

Comment (adiabramovitch) [reply internal/clients/new_instance_types_test.go:1]: Yes! It bothered me as well 😀

**Review by lzap — APPROVED** [review]
Lol the test is here.

**Pull request merged**